src/mmw/apps/geoprocessing_api/calcs.py

```diff
@@ -59,12 +59,21 @@ def stream_data(results, geojson):
     Returns a dictionary to append to outgoing JSON for analysis results.
     """
 
+    NULL_SLOPE = -9998.0
+
     sql = '''
-        SELECT sum(lengthkm) as lengthkm, stream_order
+        SELECT sum(lengthkm) as lengthkm,
+               stream_order,
+               sum(case when NULLIF(slope, {NULL_SLOPE}) is NULL
```

Comment:
This could be simplified by doing something like:

```sql
SUM(lengthkm * NULLIF(slope, {NULL_SLOPE})) AS slopesum
```

Since arithmetic with NULLs results in NULLs.